src/modules/Dashboard/pages/ParticipantActivityDetails/ParticipantActivityDetails.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useMemo } from 'react';
+import { generatePath, useNavigate, useParams } from 'react-router-dom';
+
+import { useAppDispatch } from 'redux/store';
+import { Spinner, Svg } from 'shared/components';
+import { workspaces } from 'shared/state';
+import {
+  StyledBody,
+  StyledFlexTopCenter,
+  StyledLogo,
+  StyledTitleLargish,
+  StyledHeadlineLarge,
+  theme,
+} from 'shared/styles';
+import { applet as appletState } from 'shared/state';
+import { applets, users } from 'modules/Dashboard/state';
+import { getRespondentDetails } from 'modules/Dashboard/state/Users/Users.thunk';
+import { palette } from 'shared/styles/variables/palette';
+import { page } from 'resources';
+import { getEntityKey } from 'shared/utils';
+
+export const ParticipantActivityDetails = () => {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const { appletId, participantId, activityId } = useParams();
+  const { ownerId } = workspaces.useData() || {};
+  const appletLoadingStatus = appletState.useResponseStatus();
+  const respondentLoadingStatus = users.useRespondentStatus();
+  const { useRespondent } = users;
+  const { useAppletData } = appletState;
+  const applet = useAppletData();
+  const respondent = useRespondent();
+  const { getApplet } = appletState.thunk;
+
+  useEffect(() => {
+    if (!appletId) return;
+    dispatch(getApplet({ appletId }));
+
+    if (!participantId || !ownerId) return;
+    dispatch(getRespondentDetails({ ownerId, appletId, respondentId: participantId }));
+
+    return () => {
+      dispatch(applets.actions.resetEventsData());
+    };
+  }, [appletId, participantId, ownerId]);
+
+  const navigateUp = () => {
+    navigate(
+      generatePath(page.appletParticipantActivities, {
+        participantId,
+        appletId,
+      }),
+    );
+  };
+
+  const loading =
+    appletLoadingStatus === 'loading' ||
+    appletLoadingStatus === 'idle' ||
+    respondentLoadingStatus === 'loading' ||
+    respondentLoadingStatus === 'idle';
+
+  useEffect(() => {
+    if (!loading && !respondent) {
```

Comment:
Could you please use early return to get rid of nesting?
```suggestion
    if (loading || respondent) return;
    navigateUp();
```